test/RpcEvmCompatibility.js

```diff
@@ -0,0 +1,275 @@
+
+        const txHash = await provider.send('eth_sendRawTransaction', [rawTx]);
+        const receipt = await provider.waitForTransaction(txHash);
+
+        expect(txHash).to.be.a('string');
+        expect(txHash).to.match(/0x[0-9a-fA-F]{64}/);
+        await checkReceipt(receipt, newValue);
+    });
+
+    it('should support RPC method eth_getBalance', async () => {
+        const balance = await provider.send('eth_getBalance', [signer.address, 'latest']);
+
+        expect(balance).to.be.a('string');
+        expect(BigNumber.from(balance)).to.be.gt(0);
+    });
+
+    it('should support RPC method eth_syncing', async () => {
+        const syncingStatus = await provider.send('eth_syncing', []);
+
+        if (syncingStatus) {
+            throw new Error('The provided rpc node is not synced');
+        } else {
+            expect(syncingStatus).to.be.false;
+        }
+    });
+
+    it('should support RPC method eth_subscribe', async function () {
+        // This uses eth_subscribe
+        // Setting up manually via wss rpc is tricky
+        const newValue = 500;
+        let isSubscribe = false;
+        rpcCompatibilityContract.on('ValueUpdatedForSubscribe', (value) => {
+            expect(value.toNumber()).to.equal(newValue);
+            isSubscribe = true;
+        });
+
+        await rpcCompatibilityContract.updateValueForSubscribe(newValue).then((tx) => tx.wait());
+        await waitFor(5, () => {
+            expect(isSubscribe).to.equal(true);
+        });
+    });
+
+    describe('eip-1559 supported rpc methods', () => {
+        if (!isHardhat) {
+            it('should support RPC method eth_maxPriorityFeePerGas', async () => {
+                const maxPriorityFeePerGas = await provider.send('eth_maxPriorityFeePerGas', []);
+
+                expect(maxPriorityFeePerGas).to.be.a('string');
+                expect(BigNumber.from(maxPriorityFeePerGas).toNumber()).to.be.at.least(0);
+
+                const gasLimit = network.config.gasOptions?.gasLimit || 50000;
+                const gasOptions = { maxPriorityFeePerGas, gasLimit };
+                const newValue = 600;
+                const receipt = await rpcCompatibilityContract.updateValue(newValue, gasOptions).then((tx) => tx.wait());
+                await checkReceipt(receipt, newValue);
+            });
+        }
+
+        it('should support RPC method eth_feeHistory', async () => {
+            const feeHistory = await provider.send('eth_feeHistory', ['0x1', 'latest', [25, 75]]); // reference: https://docs.alchemy.com/reference/eth-feehistory
+
+            expect(feeHistory).to.be.an('object');
+            expect(parseInt(feeHistory.oldestBlock, 16)).to.be.an('number');
+            expect(feeHistory.reward).to.be.an('array');
+
+            const gasOptions = {};
+            const baseFeePerGas = feeHistory.baseFeePerGas[0];
+            gasOptions.gasPrice = BigNumber.from(baseFeePerGas);
+            const newValue = 700;
```

Comment:
should also add a test for setting maxFeePerGas in gas options along with max priority fee based on fee history to check eip 1559 pricing